src/Command/Phar/SelfUpdateCommand.php

```diff
@@ -0,0 +1,196 @@
+    use OutputStyleAwareTrait;
+    use ConsoleAwareTrait;
+
+    private const LASTESTRELEASE_PATTERN = 'https://github.com/concrete5/console/releases/latest/download/%s';
+
+    public static function register(Container $container, Application $console): void
+    {
+        $console->command('self-update', self::class)
+            ->descriptions(
+                'Update concrete.phar to the latest version.'
+            )
+        ;
+    }
+
+    /**
+     * @throws \RuntimeException
+     */
+    public function __invoke(Application $console): int
+    {
+        $this->checkRequirements();
+        $myVersion = $this->getMyVersion();
+        $latestVersion = $this->getMostRecentAvailableVersion();
+        if (version_compare($myVersion, $latestVersion) >= 0) {
+            if (!$this->output->isQuiet()) {
+                $this->output->writeln(
+<<<EOT
+<comment>You are already using the most recent version:
+- your version: {$myVersion}
+- latest version: {$latestVersion}</comment>
+EOT
+                );
+            }
+            return 0;
+        }
+        $pharContents = $this->fetchPharContents();
+        $this->updateMyPhar($pharContents);
+        if (!$this->output->isQuiet()) {
+            $this->output->writeln(
+<<<EOT
+<info>The application has been upgraded from version {$myVersion} to {$latestVersion}</info>
+EOT
+            );
+        }
+        return 0;
+    }
+
+    private function getPharPath(): string
+    {
+        return Phar::running(false);
+    }
+
+    /**
+     * @throws \RuntimeException
+     */
+    private function checkRequirements(): void
+    {
+        $pharPath = $this->getPharPath();
+        if (!is_writeable($pharPath)) {
+            throw new RuntimeException("The file {$pharPath} can't be written: you may need to run this command as root.");
+        }
+    }
+
+    private function getMyVersion(): string
+    {
+        return $this->console->getVersion();
+    }
+
+    /**
+     * @throws \RuntimeException
+     */
+    private function getMostRecentAvailableVersion(): string
+    {
+        return trim($this->download('version.txt'));
+    }
+
+    /**
+     * @throws \RuntimeException
+     */
+    private function fetchPharContents(): string
+    {
+        $expectedSha384 = trim($this->download('concrete.sig'));
+        $phar = $this->download('concrete.phar');
+        if ($this->output->isVerbose()) {
+            $this->output->write('Checking signature of the PHAR file... ');
+        }
+        $actualSha384 = hash('sha384', $phar);
+        if ($expectedSha384 !== $actualSha384) {
+            throw new RuntimeException(
+<<<EOT
+The downloaded phar file is corrupt!
+Expected SHA-384: {$expectedSha384}
+Actual SHA-384  : {$actualSha384}
+EOT
+            );
+        }
+        if ($this->output->isVerbose()) {
+            $this->output->outputDone("passed.\n");
+        }
+
+        return $phar;
+    }
+
+    /**
+     * @throws \RuntimeException
```

Comment:
Do you mind adding a description to the throws?

Reply:
Done: f12fd856a4f0ff803dc39e8ccbc0253c0b6fe312